### ISSUE TITLE
PSO variable type buglet

### ISSUE THREAD

**Issue by coatless** [mlpack/ensmallen#123]
#### Issue description

Mismatch on return types in [PSO](https://github.com/mlpack/ensmallen/blob/b46986d1344551b1d8480784661b7c547304ba04/include/ensmallen_bits/pso/pso.hpp)

```
../inst/include/ensmallen_bits/pso/pso.hpp: In member function ‘size_t ens::PSOType<VelocityUpdatePolicy, InitPolicy>::LowerBound() const’:
../inst/include/ensmallen_bits/pso/pso.hpp:162:38: error: cannot convert ‘const vec’ {aka ‘const arma::Col<double>’} to ‘size_t’ {aka ‘long unsigned int’} in return
  162 |   size_t LowerBound() const { return lowerBound; }
      |                                      ^~~~~~~~~~
```

Defined as `arma::vec`:

https://github.com/mlpack/ensmallen/blob/b46986d1344551b1d8480784661b7c547304ba04/include/ensmallen_bits/pso/pso.hpp#L207-L210

But returning `size_t`:

https://github.com/mlpack/ensmallen/blob/b46986d1344551b1d8480784661b7c547304ba04/include/ensmallen_bits/pso/pso.hpp#L161-L171

#### Your environment

 * version of ensmallen: 1.16.0
 * operating system: debian
 * compiler: gcc-9
 * version of Armadillo: 9.600

**Comment by zoq** [comment]
Thanks for the report, should be fixed in https://github.com/mlpack/ensmallen/pull/124, I will put some time into the test setup, so that we can avoid such issues in the future.

**Comment by rcurtin** [comment]
Fixed in #124.  Thanks for the report. :+1:

**Comment by rcurtin** [comment]
Ah, I discovered that #126 will also be necessary here.

**Comment by coatless** [comment]
@rcurtin sorry about that. I should have grabbed the entire log. :/

**Comment by coatless** [comment]
Closed in #126 and #123
